wellcomeml/ml/bert_semantic_equivalence.py

```diff
@@ -0,0 +1,198 @@
+from collections import defaultdict
+import os
+
+import tensorflow as tf
+from transformers import (
+    BertConfig,
+    BertTokenizer,
+    TFBertForSequenceClassification
+)
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.model_selection import train_test_split
+
+from wellcomeml.ml.bert_vectorizer import check_cache_and_download
+
+
+class SemanticEquivalenceClassifier(BaseEstimator, TransformerMixin):
+    """
+    Class to fine-tune BERT-type models for semantic equivalence, for example
+    paraphrase, textual similarity and other NLU tasks
+    """
+    def __init__(self, pretrained='bert',
+                 batch_size=32,
+                 eval_batch_size=32*2,
+                 learning_rate=3e-5,
+                 test_size=0.2,
+                 max_length=128):
+        self.pretrained = pretrained
+        self.batch_size = batch_size
+        self.eval_batch_size = eval_batch_size
+        self.learning_rate = learning_rate
+        self.test_size = test_size
+        self.max_length = max_length
+
+        # Defines attributes that will be initialised later
+        self.config = None
+        self.tokenizer = None
+        self.model = None
+        self.train_dataset = None
+        self.valid_dataset = None
+        self.train_steps = None
+        self.valid_steps = None
+        
+        # Flag to determine whether a model was previously trained or not
+        self._trained = False
```

Comment:
possibly it would be nice if this was sklearn compatible `https://scikit-learn.org/stable/modules/generated/sklearn.utils.validation.check_is_fitted.html`, i think it would need an undescore in the end.

Reply:
That is a really good suggestion - I will look into it before merging.

Reply:
I implemented this in https://github.com/wellcometrust/WellcomeML/pull/43/commits/750e13a46fde40a2b3c94f3334e837b9d7537b2f